proparse/src/main/java/org/prorefactor/treeparser/ParseUnit.java

```diff
@@ -114,8 +114,13 @@ public File getFile() {
     return MacroLevel.sourceArray(macroGraph);
   }
 
-  public TokenStream lex() throws RefactorException {
-    LOGGER.trace("Entering ParseUnit#lex()");
+  public TokenStream lex() throws IOException, RefactorException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.io.IOException, org.prorefactor.refactor.RefactorException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1160)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.prorefactor.refactor.RefactorException', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)